### ISSUE TITLE
Subprocesses killed after accelerate launch

### ISSUE THREAD

**Issue by vblagoje** [huggingface/accelerate#193]
I am running distributed training on a single node with 4 GPUs. This is not my first time using accelerate in a multi GPU environment, but this training is different because a large JSON file is loaded before the train loop. Loading the file takes about 90 sec or so. As the loading gets completed, I get the following stack trace:

```
INFO:__main__:Distributed environment: MULTI_GPU  Backend: nccl
Num processes: 4
Process index: 0
Local process index: 0
Device: cuda:0
Use FP16 precision: False

INFO:__main__:Loading prepared training docs from ./eli5_train_precomputed_dense_docs.json
Killing subprocess 5571
Killing subprocess 5572
Killing subprocess 5573
Killing subprocess 5574
Traceback (most recent call last):
  File "/usr/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/home/vblagoje/eli5/lib/python3.8/site-packages/torch/distributed/launch.py", line 340, in <module>
    main()
  File "/home/vblagoje/eli5/lib/python3.8/site-packages/torch/distributed/launch.py", line 326, in main
    sigkill_handler(signal.SIGTERM, None)  # not coming back
  File "/home/vblagoje/eli5/lib/python3.8/site-packages/torch/distributed/launch.py", line 301, in sigkill_handler
    raise subprocess.CalledProcessError(returncode=last_return_code, cmd=cmd)
subprocess.CalledProcessError: Command '['/home/vblagoje/eli5/bin/python3', '-u', 'run_seq2seq_no_trainer.py', '--pad_to_max_length']' died with <Signals.SIGKILL: 9>.
```

Any ideas on how to deal with this? I suspect I need to increase some timeout somewhere?

**Comment by sgugger** [comment]
The process are automatically killed by the PyTorch launcher when one of them dies, to avoid getting zombies when an error is raised. The timeout in the launcher is 30 minutes by default, so your 90s don't seem to be relevant. Maybe one of the processes goes out of RAM?

**Comment by vblagoje** [comment]
Right, forgot loading the same file in four different processes might cause that. Closing it now. Thank you @sgugger

**Comment by neel04** [comment]
Any easy way to resolve that? I'm using a numpy-based iterator and running out of RAM when using all processes/GPUs on my machine @sgugger;
